### PR TITLE
refactor!: remove deprecated SysMonitor

### LINKED ISSUE #10615: SysMonitor is throwing 'SigarFileNotFoundException' in official Docker images
We can see many instances of `org.hyperic.sigar.SigarFileNotFoundException: No such file or directory` in the log file after adding `org.apache.druid.java.util.metrics.SysMonitor` to the list of monitors.
### Affected Version
We are using the official helm chart "incubator/druid" version "0.2.18

### Description

Please include as much detailed information about the problem as possible.
We are using the helm chart as is while adding some monitors such as the `org.apache.druid.java.util.metrics.SysMonitor`.
- Cluster size: Default
- Configurations in use: Additional Monitors
- Steps to reproduce the problem: Just run the Helm chart and run some ingestion. We specifically use 'kafka_ingestion"

Some log file excerpts:
`2020-10-18T12:16:57,518 ERROR [MonitorScheduler-0] org.apache.druid.java.util.metrics.SysMonitor - Failed to get DiskUsage[/opt/apache-druid-0.19.0/var]
org.hyperic.sigar.SigarFileNotFoundException: No such file or directory
	at org.hyperic.sigar.DiskUsage.gather(Native Method) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.DiskUsage.fetch(DiskUsage.java:30) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.Sigar.getDiskUsage(Sigar.java:680) ~[sigar-1.6.5.132.jar:?]
	at org.apache.druid.java.util.metrics.SysMonitor$DiskStats.emit(SysMonitor.java:310) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.SysMonitor.doMonitor(SysMonitor.java:107) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.AbstractMonitor.monitor(AbstractMonitor.java:47) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:124) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:119) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:158) [druid-core-0.19.0.jar:0.19.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_252]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
2020-10-18T12:17:57,517 ERROR [MonitorScheduler-0] org.apache.druid.java.util.metrics.SysMonitor - Failed to get DiskUsage[/var/druid]
org.hyperic.sigar.SigarFileNotFoundException: No such file or directory
	at org.hyperic.sigar.DiskUsage.gather(Native Method) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.DiskUsage.fetch(DiskUsage.java:30) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.Sigar.getDiskUsage(Sigar.java:680) ~[sigar-1.6.5.132.jar:?]
	at org.apache.druid.java.util.metrics.SysMonitor$DiskStats.emit(SysMonitor.java:310) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.SysMonitor.doMonitor(SysMonitor.java:107) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.AbstractMonitor.monitor(AbstractMonitor.java:47) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:124) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:119) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:158) [druid-core-0.19.0.jar:0.19.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_252]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
2020-10-18T12:17:57,518 ERROR [MonitorScheduler-0] org.apache.druid.java.util.metrics.SysMonitor - Failed to get DiskUsage[/dev/termination-log]
org.hyperic.sigar.SigarFileNotFoundException: No such file or directory
	at org.hyperic.sigar.DiskUsage.gather(Native Method) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.DiskUsage.fetch(DiskUsage.java:30) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.Sigar.getDiskUsage(Sigar.java:680) ~[sigar-1.6.5.132.jar:?]
	at org.apache.druid.java.util.metrics.SysMonitor$DiskStats.emit(SysMonitor.java:310) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.SysMonitor.doMonitor(SysMonitor.java:107) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.AbstractMonitor.monitor(AbstractMonitor.java:47) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:124) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:119) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:158) [druid-core-0.19.0.jar:0.19.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_252]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
2020-10-18T12:17:57,518 ERROR [MonitorScheduler-0] org.apache.druid.java.util.metrics.SysMonitor - Failed to get DiskUsage[/etc/resolv.conf]
org.hyperic.sigar.SigarFileNotFoundException: No such file or directory
	at org.hyperic.sigar.DiskUsage.gather(Native Method) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.DiskUsage.fetch(DiskUsage.java:30) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.Sigar.getDiskUsage(Sigar.java:680) ~[sigar-1.6.5.132.jar:?]
	at org.apache.druid.java.util.metrics.SysMonitor$DiskStats.emit(SysMonitor.java:310) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.SysMonitor.doMonitor(SysMonitor.java:107) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.AbstractMonitor.monitor(AbstractMonitor.java:47) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:124) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:119) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:158) [druid-core-0.19.0.jar:0.19.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_252]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
2020-10-18T12:17:57,518 ERROR [MonitorScheduler-0] org.apache.druid.java.util.metrics.SysMonitor - Failed to get DiskUsage[/etc/hostname]
org.hyperic.sigar.SigarFileNotFoundException: No such file or directory
	at org.hyperic.sigar.DiskUsage.gather(Native Method) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.DiskUsage.fetch(DiskUsage.java:30) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.Sigar.getDiskUsage(Sigar.java:680) ~[sigar-1.6.5.132.jar:?]
	at org.apache.druid.java.util.metrics.SysMonitor$DiskStats.emit(SysMonitor.java:310) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.SysMonitor.doMonitor(SysMonitor.java:107) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.AbstractMonitor.monitor(AbstractMonitor.java:47) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:124) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:119) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:158) [druid-core-0.19.0.jar:0.19.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_252]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
2020-10-18T12:17:57,518 ERROR [MonitorScheduler-0] org.apache.druid.java.util.metrics.SysMonitor - Failed to get DiskUsage[/etc/hosts]
org.hyperic.sigar.SigarFileNotFoundException: No such file or directory
	at org.hyperic.sigar.DiskUsage.gather(Native Method) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.DiskUsage.fetch(DiskUsage.java:30) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.Sigar.getDiskUsage(Sigar.java:680) ~[sigar-1.6.5.132.jar:?]
	at org.apache.druid.java.util.metrics.SysMonitor$DiskStats.emit(SysMonitor.java:310) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.SysMonitor.doMonitor(SysMonitor.java:107) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.AbstractMonitor.monitor(AbstractMonitor.java:47) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:124) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:119) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:158) [druid-core-0.19.0.jar:0.19.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_252]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
2020-10-18T12:17:57,518 ERROR [MonitorScheduler-0] org.apache.druid.java.util.metrics.SysMonitor - Failed to get DiskUsage[/opt/apache-druid-0.19.0/var]
org.hyperic.sigar.SigarFileNotFoundException: No such file or directory
	at org.hyperic.sigar.DiskUsage.gather(Native Method) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.DiskUsage.fetch(DiskUsage.java:30) ~[sigar-1.6.5.132.jar:?]
	at org.hyperic.sigar.Sigar.getDiskUsage(Sigar.java:680) ~[sigar-1.6.5.132.jar:?]
	at org.apache.druid.java.util.metrics.SysMonitor$DiskStats.emit(SysMonitor.java:310) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.SysMonitor.doMonitor(SysMonitor.java:107) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.AbstractMonitor.monitor(AbstractMonitor.java:47) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:124) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.metrics.MonitorScheduler$1.call(MonitorScheduler.java:119) [druid-core-0.19.0.jar:0.19.0]
	at org.apache.druid.java.util.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:158) [druid-core-0.19.0.jar:0.19.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_252]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]`

A possible solution might be hidden in this issue: https://itectec.com/ubuntu/ubuntu-install-cgconfig-in-ubuntu-16-04/

### PR DESCRIPTION
**Pull request by FrankChen021**
Fixes #10615.

### Description

The deprecated `org.apache.druid.java.util.metrics.SysMonitor` monitor is removed. Its no-op variant, unused Druid subclass, Guice provider, and obsolete tests are removed as well.

`JvmCpuMonitor` now uses OSHI process snapshots to report the existing total, user, system, and percentage CPU metrics. `JvmPidDiscoverer` uses Java `ProcessHandle` to discover the current JVM PID. The Sigar PID discoverer and utility, native library packaging, Maven dependencies, and associated tests have been removed.

Existing configurations that list `SysMonitor` in `druid.monitoring.monitors` must be updated to use `org.apache.druid.java.util.metrics.OshiSysMonitor`.

The metrics documentation now documents only `OshiSysMonitor`.

#### Release note

`SysMonitor` has been removed after being deprecated in Druid 27. Use `OshiSysMonitor` instead.

#### Validation

- `mvn -ntp test -pl processing,server -am -Dtest="org.apache.druid.java.util.metrics.JvmCpuMonitorTest,org.apache.druid.java.util.metrics.JvmPidDiscovererTest,org.apache.druid.java.util.metrics.OshiSysMonitorTest,org.apache.druid.java.util.metrics.NoopOshiSysMonitorTest,org.apache.druid.server.metrics.MetricsModuleTest" -Dsurefire.failIfNoSpecifiedTests=false -Pskip-static-checks -Dweb.console.skip=true -T1C`
- `npm run spellcheck`

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/java/util/metrics/JvmCpuMonitor.java [hunk 1]
```diff
@@ -19,60 +19,74 @@
...
+    final OSProcess currentProcess = operatingSystem.getCurrentProcess();
+    if (currentProcess == null) {
+      log.error("Unable to get current process CPU metrics");
+      return true;
+    }
+
+    final ServiceMetricEvent.Builder builder = builder();
+    final long userTime = currentProcess.getUserTime();
+    final long sysTime = currentProcess.getKernelTime();
+    final Map<String, Long> procDiff = diff.to(
```
Comment (FrankChen021) [inline processing/src/main/java/org/apache/druid/java/util/metrics/JvmCpuMonitor.java:75]: [P2] Do not advance the diff on failed OSHI snapshots

`OperatingSystem.getCurrentProcess()` is non-null by contract: when the native lookup fails it returns a `CurrentProcessStub` with zero or unknown fields. Because this code only checks for null, that stub is fed into `KeyedDiff` and then stored as `previousProcess`, emitting bogus zero deltas and making the next successful sample compare against the zero snapshot. Detect an invalid snapshot and leave the diff and previous-process state unchanged when OSHI cannot read the process.

**Review by FrankChen021 — COMMENTED** [review]
| Severity | Findings |
| --- | ---: |
| P0 | 0 |
| P1 | 0 |
| P2 | 1 |
| P3 | 0 |
| Total | 1 |

| Severity | Findings |
| --- | ---: |
| P0 | 0 |
| P1 | 0 |
| P2 | 1 |
| P3 | 0 |
| Total | 1 |

Reviewed 19 of 19 changed files.

---

This is an automated review by Codex GPT-5.6-Luna(max)

File: processing/src/main/java/org/apache/druid/java/util/metrics/JvmCpuMonitor.java (hunk 1 above)
Comment (FrankChen021) [reply processing/src/main/java/org/apache/druid/java/util/metrics/JvmCpuMonitor.java:75]: Addressed in the latest commit. `JvmCpuMonitor` now uses the nullable `OperatingSystem.getProcess(pid)` API, so a failed lookup is skipped before either `KeyedDiff` or `previousProcess` is updated. This prevents failed snapshots from producing bogus deltas or poisoning the next CPU percentage calculation.

Added a regression test covering a valid snapshot, a failed lookup, and the next valid snapshot; the original baselines are preserved.

Validation: 16 focused processing tests passed; Checkstyle reported 0 violations.

Reviewed 2 of 2 affected files.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 19 of 19 changed files.

Validation: `git diff --check` passed. Builds and tests were not run.



---

This is an automated review by Codex GPT-5.6-Luna(max)

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 19 of 19 changed files.

Validation: `git diff --check` passed. Builds and tests were not run.



---

This is an automated review by Codex GPT-5.6-Luna(max)

**Review by GWphua — APPROVED** [review]
